src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -53,27 +53,38 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $legacy = $parameters['legacy'];
             unset($parameters['legacy']);
         }
+        $emptyParams = empty($parameters);
+        $isSprintf = $this->isSprintfString($id);
 
         if (empty($locale)) {
             $locale = null;
         }
 
-        $translated = parent::trans($id, [], $this->normalizeDomain($domain), $locale);
+        if (!$emptyParams && $isSprintf) {
+            $translated = parent::trans(
+                $id,
+                 [],
+                $this->normalizeDomain($domain),
+                $locale
+            );
+        } else {
+            $translated = parent::trans(
+                $id,
+                $parameters,
+                $this->normalizeDomain($domain),
+                $locale
+            );
+        }
```

Comment:
```suggestion
        $translated = parent::trans($id, $isSprintf ? [] : $parameters, $this->normalizeDomain($domain), $locale);
```

Reply:
This is equivalent and much smaller. However, I'm still not very confident about simply removing the parameters from the Symfony call, because as we discovered parameters are also required to be able to handle the plural wordings correctly.

This can be acceptable if and only if the messages using the sprintf format are guaranteed to be legacy Core wordings only.